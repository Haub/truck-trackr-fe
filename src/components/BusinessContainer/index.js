import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';
import { changeCurrentPage } from '../../actions';

export const BusinessContainer = (props) => {
  const {data, changeCurrentPage} = props

  const businesses = data.map(business => 
    <NavLink 
      to={`/business/${business.attributes.name}`} 
      key={business.id} 
      onClick={()=> changeCurrentPage(business.attributes.name)}
    >
      {business.attributes.name}
    </NavLink>
  )

  return (
    <div>
      { businesses }
    </div>   
  )
  
};


export default (BusinessContainer);