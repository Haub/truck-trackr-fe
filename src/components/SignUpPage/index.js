import React, { Component } from 'react';
import { FirebaseContext, withFirebase } from '../Firebase';
import { Login } from '../../containers/Login'
import { withRouter } from 'react-router-dom'


export const SignUpPage = (props) => (
  <div>
    <FirebaseContext.Consumer>
      {firebase => <Login firebase={firebase} history={props.history}/>}
    </FirebaseContext.Consumer>
  </div>
);

export default withRouter(withFirebase(SignUpPage))