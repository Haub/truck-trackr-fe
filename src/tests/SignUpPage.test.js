import React from "react";
import { shallow } from "enzyme";
import { SignUpPage } from "../components/SignUpPage/";

describe("SIGNUPPAGE", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<SignUpPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
