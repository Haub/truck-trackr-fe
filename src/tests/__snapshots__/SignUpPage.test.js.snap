// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIGNUPPAGE should match the snapshot 1`] = `
<div>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</div>
`;
